src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/MultiSourcePackageFeed.cs

```diff
@@ -174,6 +242,24 @@ public async Task<SearchResult<IPackageSearchMetadata>> RefreshSearchAsync(Refre
                 }
             }
 
+            if (_telemetryService != null
+                && aggregated.SourceSearchStatus != null
+                && aggregated.SourceSearchStatus.Values != null
+                && telemetryState != null)
+            {
+                var loadingStatus = aggregated.SourceSearchStatus.Values.Aggregate();
+                if (loadingStatus != LoadingStatus.Loading
+                    && telemetryState.ShouldEmit)
+                {
+                    _telemetryService.EmitTelemetryEvent(new SearchPageTelemetryEvent(
+                        telemetryState.OperationId,
+                        telemetryState.PageIndex,
+                        aggregated.Items?.Count ?? 0,
+                        telemetryState.Duration,
```

Comment:
where are you stopping the watch to measure duration?

Reply:
I am capturing the duration only once so technically I don't need to stop the stopwatch. I'll add it though to be more clear.